File: frontend/src/app/shared/service/admin/common/ContactAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ContactDto} from 'src/app/shared/model/common/Contact.model';
import {ContactCriteria} from 'src/app/shared/criteria/common/ContactCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ContactAdminService extends AbstractService<ContactDto, ContactCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ContactDto {
        return new ContactDto();
    }

    public constrcutCriteria(): ContactCriteria {
        return new ContactCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/contact/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/common/ContactAdminService.java
package ma.zs.alc.service.facade.admin.common;

import java.util.List;
import ma.zs.alc.bean.core.common.Contact;
import ma.zs.alc.dao.criteria.core.common.ContactCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ContactAdminService extends  IService<Contact,ContactCriteria>  {

    List<Contact> findByCollaboratorId(Long id);
    int deleteByCollaboratorId(Long id);
    long countByCollaboratorId(Long id);




}
